src/Relation/Binary/Construct/Interior/Symmetric.agda

```diff
@@ -0,0 +1,91 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Symmetric interior of a binary relation
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Relation.Binary.Construct.Interior.Symmetric where
+
+open import Function.Base using (flip)
+open import Level
+open import Relation.Binary
+
+private
+  variable
+    a b c ℓ r s t : Level
+    A : Set a
+    R : Rel A r
+    S : Rel A s
+    T : Rel A t
+
+------------------------------------------------------------------------
+-- Definition
+
+record SymInterior (R : Rel A ℓ) (x y : A) : Set ℓ where
+  constructor _,_
+  field
+    lhs≤rhs : R x y
+    rhs≤lhs : R y x
+open SymInterior public
+
+------------------------------------------------------------------------
+-- Properties
+
+-- The symmetric interior is symmetric.
+symmetric : Symmetric (SymInterior R)
+symmetric (r , r′) = r′ , r
+
+-- The symmetric interior of R is greater than (or equal to) any other symmetric
+-- relation contained by R.
+unfold : Symmetric S → S ⇒ R → S ⇒ SymInterior R
+unfold sym f s = f s , f (sym s)
+
+-- SymInterior preserves various properties.
+reflexive : Reflexive R → Reflexive (SymInterior R)
+reflexive refl = refl , refl
+
+transitive′ : Trans R S T → Trans S R T →
+  Trans (SymInterior R) (SymInterior S) (SymInterior T)
+transitive′ trans trans′ (r , r′) (s , s′) = trans r s , trans′ s′ r′
+
+transitive : Transitive R → Transitive (SymInterior R)
+transitive {R = R} tr = transitive′ {R = R} tr tr
+
+-- The symmetric interior of a strict relation is empty.
+Empty-SymInterior : Asymmetric R → Empty (SymInterior R)
+Empty-SymInterior asym (r , r′) = asym r r′
+
+-- A reflexive transitive relation _≤_ gives rise to a poset in which the
+-- equivalence relation is SymInterior _≤_.
+record IsProset {a ℓ} {A : Set a} (≤ : Rel A ℓ) : Set (a ⊔ ℓ) where
+  field
+    refl : Reflexive ≤
+    trans : Transitive ≤
+
+record Proset c ℓ : Set (suc (c ⊔ ℓ)) where
+  infix 4 _≤_
+  field
+    Carrier : Set c
+    _≤_ : Rel Carrier ℓ
+    isProset : IsProset _≤_
+
+IsProset⇒IsPartialOrder : ∀ {a ℓ} {A : Set a} {≤ : Rel A ℓ} →
+  IsProset ≤ → IsPartialOrder (SymInterior ≤) ≤
+IsProset⇒IsPartialOrder {≤ = ≤} isProset = record
```

Comment:
As with other defns inthe `Construct` hierarchy, I think that this would conventionally simply be called `isPartialOrder`, being the proof (in lower-camel-case) of the corresponding `Structure` property (in upper-camel-case`, with the ambient `IsProset⇒` being given implicitly by module-level scoping...

Reply:
Agree that `ProSet`-related definitions belong 'somewhere else', except that, as you say, best if they don't... for now ;-)